WalletWasabi.Gui/ViewModels/TextResourceViewModelBase.cs

```diff
@@ -30,28 +34,72 @@ public string Text
 			set => this.RaiseAndSetIfChanged(ref _text, value);
 		}
 
-		public Uri Target { get; }
+		public bool EmptyContent
+		{
+			get => _emptyContent;
+			set => this.RaiseAndSetIfChanged(ref _emptyContent, value);
+		}
+
+		public TextResource TextResource { get; }
 
 		private async Task<string> LoadDocumentAsync(Uri target)
 		{
 			var assetLocator = AvaloniaLocator.Current.GetService<IAssetLoader>();
-
 			using var stream = assetLocator.Open(target);
 			using var reader = new StreamReader(stream);
 			return await reader.ReadToEndAsync();
 		}
 
+		private static async Task<string> LoadDocumentAsync(string path)
```

Comment:
Just a question:
`LoadDocumentAsync` or
`LoadDocumentsAsync`

Reply:
I'd go with singular, because it's one document, although 3 different content, idk, doesn't matter. Both can work.